windows-dev.md

```diff
@@ -0,0 +1,112 @@
+# Developing on a Windows machine
+
+Sadly, many devs think that real software development on Windows is either an oxymoron
+or just for finance companies paying tons of money for MSDN licenses.
+
+This is wildly untrue: I do all of my personal dev (which is extensive) on a Windows machine.
+
+In the past, all my dev was in a Linux desktop/GUI OS in Virtualbox
+but Windows itself has come a long, long way just in the past year and some.
+
+(These days I do all my dev inside vagrant in virtualbox, using cygwin bash in windows terminal
+-- usually with docker apps running inside all of that. But I'm excited for
+[WSL2](https://www.hanselman.com/blog/HowToSetUpDockerWithinWindowsSystemForLinuxWSL2OnWindows10.aspx) to come out!)
+
+## Windows dev options
+
+Running Linux directly in Virtualbox, if you are familiar with Linux,
+is the easiest ([unless you have Docker for Windows installed!](
+https://nickjanetakis.com/blog/docker-tip-13-get-docker-for-windows-and-virtualbox-working-together)),
+but also the most effortful and the biggest shift of environment.
+
+Otherwise you will be running in the Windows environment so
+you will always want to install the new Microsoft Windows Terminal application,
+almost certainly via Chocolatey (if you don't already have that):
+
+### Install Chocolatey
+
+https://chocolatey.org/install
+
+Run Powershell as Administrator and in that:
+
+    Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+
+### Install additional tools:
+
+Research each of these if you are unfamiliar -- and I highly recommend you become familiar with all of them!
+
+    choco upgrade -y microsoft-windows-terminal cygwin virtualbox vagrant
+
+You might also want to install VS Code, or Jetbrains' IntelliJ, RubyMine, PyCharm, or WebStorm -- or any other IDEs of your choice.
+For example:
+
+    choco upgrade -y vscode intellijidea-community rubymine pycharm-community webstorm
+
+## Setting up Windows dev envs
+
+There are 6 main options, installation instructions follow -- make sure you can run a basic Hello World app in each:
+
+#### 1. In Windows itself:
+
+    choco upgrade -y ruby python nodejs ojdkbuild dotnetcore-sdk golang
+
+Then you can run those languages in a cmd/powershell shell in Windows Terminal.
```

Comment:
[MinGW](https://chocolatey.org/packages/mingw) is worth mentioning, for C and C++ developers who want to use GCC as their compiler. I've been using it for years with much success.

[MSYS/MSYS2](https://chocolatey.org/packages/msys2) are also worth mentioning. They're essentially just a straight port of a _lot_ of common Unix tools (e.g. Make, bash, grep, etc.) to Windows, which is great if you want those tools but you don't want to deal with Cygwin/WSL/etc.